pom.xml

```diff
@@ -100,9 +99,30 @@
           <compatibleSinceVersion>1.3.0</compatibleSinceVersion>
         </configuration>
       </plugin>
+
+      <plugin>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>display-info</id>
+            <configuration>
+              <rules>
+                <requireUpperBoundDeps>
+                  <excludes combine.children="append">
+                    <!--for org.jenkins-ci.plugins:matrix-auth:1.16-->
+                    <exclude>org.jenkins-ci.plugins.icon-shim:icon-set</exclude>
+                    <!--for org.jenkins-ci.plugins:junit:1.21-->
+                    <exclude>org.jenkins-ci:annotation-indexer</exclude>
```

Comment:
This is not that fine. It should have blown up in https://github.com/jenkinsci/junit-plugin/pull/80, CC @abayer and @varyvol IIUC.

Reply:
At a minimum this should rather be done by `exclude`ing that transitive dep from `junit`.

Reply:
Put the two dependencies that cause problems in local exclusions instead of using appender on the maven-enforcer-plugin